base/src/main/kotlin/net/corda/v5/base/types/MemberX500Name.kt

```diff
@@ -16,32 +16,120 @@ import javax.security.auth.x500.X500Principal
+ * ```
+ *
+ * ```kotlin
+ * val commonName = "commonName"
+ * val organisationUnit = "organisationUnit"
+ * val organization = "organization"
+ * val locality = "London"
+ * val state = "state"
+ * val country = "GB"
+ *
+ * val exampleNameFirst = MemberX500Name(organization, locality, country)
+ * val exampleNameSecond = MemberX500Name(commonName, organisationUnit, organization, locality, state, country)
+ * val exampleNameThird = MemberX500Name(commonName, organization, locality, country)
+ *
+ * val commonNameForExampleNameThird = exampleNameThird.commonName
+ * val organisationUnitForExampleNameThird = exampleNameThird.organisationUnit
+ * val organisationForExampleNameThird = exampleNameThird.organisation
+ * val localityForExampleNameThird = exampleNameThird.locality
+ * val stateForExampleNameThird = exampleNameThird.state
+ * val countryForExampleNameThird = exampleNameThird.country
+ * val principalForExampleNameThird = exampleNameThird.x500Principal
+ * val name = "O=organization,L=London,C=GB"
+ * val principalForNewName = X500Principal(name)
+ * val nameByPrincipal = MemberX500Name.build(principalForNewName)
+ * val nameByParse = MemberX500Name.parse(name)
+ *
+ * val map = MemberX500Name.toAttributesMap("CN=alice, OU=Accounting, O=R3, L=Seattle, ST=Washington, C=US")
+ * ```
+ *
+ * @author Alexey Kadyrov
+ * @since DP2
+ *
+ * @param commonName Optional name by which the entity is usually known. Used only for services (for
  * organisations, the [organisation] property is the name). Corresponds to the "CN" attribute type.
- * @property organisationUnit optional name of a unit within the [organisation]. Corresponds to the "OU" attribute type.
- * @property organisation name of the organisation. Corresponds to the "O" attribute type.
- * @property locality locality of the organisation, typically the nearest major city. For distributed services this would be
+ * @param organisationUnit Optional name of a unit within the [organisation]. Corresponds to the "OU" attribute type.
+ * @param organisation Name of the organisation. Corresponds to the "O" attribute type.
+ * @param locality Locality of the organisation, typically the nearest major city. For distributed services this would be
  * where one of the organisations is based. Corresponds to the "L" attribute type.
- * @property state the full name of the state or province the organisation is based in. Corresponds to the "ST"
+ * @param state The full name of the state or province the organisation is based in. Corresponds to the "ST"
  * attribute type.
- * @property country country the organisation is in, as an ISO 3166-1 2-letter country code. Corresponds to the "C"
+ * @param country Country the organisation is in, as an ISO 3166-1 2-letter country code. Corresponds to the "C"
  * attribute type.
 */
 @Suppress("LongParameterList")
 @CordaSerializable
 class MemberX500Name(
+    /**
```

Comment:
Out of curiosity why the @property where changed to @param ?

Reply:
They are still listed as properties as well, by putting the comments above each fields.